test/LAMMPS_jll/.pkg/platform_augmentation.jl

```diff
@@ -19,11 +20,23 @@ function set_abi(abi)
     end
     set_preferences!(LAMMPS_UUID, "abi" => string(abi), force=true)
     @warn "The MPI abi has changed, you will need to restart Julia for the change to take effect" abi
+    # Note: We either need to use `LazyArtifacts` or at this point ensure that we download artifacts
+    #       for the new ABI.
+    # Issues: On 1.7 this seems to not work since while we are calling "select_artifacts.jl" we somewhere
+    #         discard the platform we pass to instantiate
+    #         On 1.6 "select_artifacts.jl" does not exist, and thus on first call we might have choosen the
+    #         wrong artifact to begin with an error there.
+    let platform = HostPlatform()
+        platform["mpi"] = string(abi)
+        Pkg.API.instantiate(; platform, verbose=true)
+    end
     return abi
 end
 
 function augment_platform!(platform)
```

Comment:
Note to self is select_platform.jl executed with the right LocalPreferences.toml?

Reply:
I think this is potentially a big problem. :(

Reply:
Yeah this is was why I noticed https://github.com/JuliaLang/Pkg.jl/pull/2901

Reply:
It's basically a homomorph of https://github.com/JuliaLang/Pkg.jl/issues/2500